cleanlab/multiannotator.py

```diff
@@ -0,0 +1,859 @@
+                    [
+                        np.mean(
+                            annotator_agreement_with_consensus[
+                                labels_multiannotator.iloc[idx][
+                                    labels_multiannotator.iloc[idx] == label
+                                ].index.values
+                            ]
+                        )
+                        for label in label_mode
+                    ]
+                )
+                max_score = np.where(label_quality_score == label_quality_score.max())[0]
+                if len(max_score) == 1:
+                    consensus_label[idx] = label_mode[max_score[0]]
+                    del tied_idx[idx]
+                else:
+                    tied_idx[idx] = label_mode[max_score]
+
+        # if still tied, break by random selection
+        if len(tied_idx) > 0:
+            warnings.warn(
+                f"breaking ties of examples {list(tied_idx.keys())} by random selection, you may want to set seed for reproducability"
+            )
+            for idx, label_mode in tied_idx.items():
+                consensus_label[idx] = np.random.choice(label_mode)
+
+    else:
+        raise ValueError(
+            f"""
+            {consensus_method} is not a valid consensus method!
+            Please choose a valid consensus_method: {valid_methods}
+            """
+        )
+
+    return consensus_label.astype("int64")
+
+
+def _get_annotator_agreement(
+    labels_multiannotator: pd.DataFrame,
+    consensus_label: np.ndarray,
+) -> np.ndarray:
+    """Returns the fractions of annotators that agree with the consensus labels. Note that the
+    fraction for each example only considers the annotators that labeled that particular example.
+
+    Parameters
+    ----------
+    labels_multiannotator : pd.DataFrame
+        2D pandas DataFrame of multiple given labels for each example with shape (N, M),
+        where N is the number of examples and M is the number of annotators.
+        For more details, labels in the same format expected by the :py:func:`get_label_quality_multiannotator <cleanlab.multiannotator.get_label_quality_multiannotator>`.
+    consensus_label : np.ndarray
+        An array of shape ``(N,)`` with the consensus labels aggregated from all annotators.
+
+    Returns
+    -------
+    annotator_agreement : np.ndarray
+        An array of shape ``(N,)`` with the fraction of annotators that agree with each consensus label.
+    """
+    annotator_agreement = labels_multiannotator.assign(consensus_label=consensus_label).apply(
+        lambda s: np.mean(s.drop("consensus_label").dropna() == s["consensus_label"]),
+        axis=1,
+    )
+    return annotator_agreement.to_numpy()
+
+
+def _get_annotator_agreement_with_annotators(
+    labels_multiannotator: pd.DataFrame,
+    num_annotations: np.ndarray,
+) -> np.ndarray:
+    """Returns the average agreement of each annotators with other annotatos that label the same examples.
```

Comment:
Returns the average agreement of each annotator with other annotators that label the same example.

Reply:
(bad spelling/grammar ...) 
Thanks for catching! Edits made